external-crates/move/crates/move-trace-format/src/trace_format.rs

```diff
@@ -0,0 +1,387 @@
+            pc,
+            gas_left,
+            instruction: format!("{:?}", instruction_opcode(instruction)),
+        });
+        for effect in effects {
+            self.push_event(TraceEvent::Effect(effect));
+        }
+    }
+
+    /// Push an `Effect` event to the trace.
+    pub fn effect(&mut self, effect: Effect) {
+        self.push_event(TraceEvent::Effect(effect));
+    }
+
+    // All events pushed to the trace are first pushed, and then the tracer is notified of the
+    // event.
+    fn push_event(&mut self, event: TraceEvent) {
+        self.trace.events.push(event.clone());
+        self.tracer.notify(&event, Writer(&mut self.trace));
+    }
+}
+
+impl Display for TraceValue {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            TraceValue::RuntimeValue { value } => {
+                write!(f, "{:#}", value)
+            }
+            TraceValue::ImmRef { location, snapshot } => {
+                write!(f, "&({}) {}", location, snapshot)
+            }
+            TraceValue::MutRef { location, snapshot } => {
+                write!(f, "&mut({}) {}", location, snapshot)
+            }
+        }
+    }
+}
+
+impl Display for Location {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Location::Local(frame_idx, idx) => {
+                write!(f, "l{idx}@{frame_idx}")
+            }
+            Location::Indexed(loc, offset) => {
+                write!(f, "({loc})[{offset}]")
+            }
+            Location::Global(id) => {
+                write!(f, "g{}", id)
+            }
+        }
+    }
+}
+
+impl Display for Effect {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Effect::Pop(value) => {
+                write!(f, "Pop {}", value)
+            }
+            Effect::Push(value) => {
+                write!(f, "Push {}", value)
+            }
+            Effect::Read(Read {
+                location,
+                root_value_read: value_read,
+                moved,
+            }) => {
+                let arrow = if *moved { "==>" } else { "-->" };
+                write!(f, "{location} {arrow} {value_read}")
+            }
+            Effect::Write(Write {
+                location,
+                root_value_after_write: value_written,
+            }) => {
+                write!(f, "{location} <- {value_written}")
+            }
+            Effect::ExecutionError(error_string) => {
+                write!(f, "ExecutionError: {error_string}")
+            }
+            Effect::DataLoad(DataLoad {
+                ref_type,
+                location,
+                snapshot,
+            }) => {
+                let ref_type = match ref_type {
+                    RefType::Imm => "&",
+                    RefType::Mut => "&mut",
+                };
+                write!(f, "G{ref_type}{location} ~~> {snapshot}")
```

Comment:
That it is a global reference that is being loaded (i.e., a reference from a native) as opposed to a "normal" reference. However, since `Global` is a location type now this isn't needed I believe. I will take a pass through the display impls for this generally :)